functions-python/update_validation_report/src/main.py

```diff
@@ -0,0 +1,187 @@
+from time import sleep
+from typing import List
+
+import functions_framework
+import requests
+import sqlalchemy.orm
+from google.cloud import storage
+from sqlalchemy.engine import Row
+from sqlalchemy.engine.interfaces import Any
+
+from database_gen.sqlacodegen_models import Gtfsdataset, Gtfsfeed
+from helpers.database import start_db_session
+from helpers.logger import Logger
+
+logging.basicConfig(level=logging.INFO)
+
+
+@functions_framework.http
+def update_validation_report(_):
+    """
+    Update the validation report for the datasets that need it
+    """
+    Logger.init_logger()
+    env = os.getenv("ENV", "dev").lower()
+
+    # Get validator version
+    validator_version = get_validator_version()
+
+    bucket_name = f"mobilitydata-datasets-{env}"
+    logging.info(f"Accessing bucket {bucket_name}")
+
+    session = start_db_session(os.getenv("FEEDS_DATABASE_URL"), echo=False)
+    latest_datasets = get_latest_datasets(session)
+    logging.info(f"Retrieved {len(latest_datasets)} latest datasets.")
+
+    dataset_blobs = get_dataset_blobs_for_validation(
+        bucket_name, validator_version, latest_datasets
+    )
+    logging.info(f"Retrieved {len(dataset_blobs)} blobs to update.")
+
+    updated_datasets = update_dataset_metadata(dataset_blobs, validator_version)
+    response = {
+        "message": f"Updated {len(updated_datasets)} validation report(s).",
+        "updated_datasets": sorted(updated_datasets),
+        "ignored_datasets": sorted(
+            [
+                dataset_id
+                for _, dataset_id in latest_datasets
+                if dataset_id not in updated_datasets
+            ]
+        ),
+    }
+    return response, 200
+
+
+def get_validator_version():
+    """
+    Get the version of the validator
+    :return: the version of the validator
+    """
+    web_validator_endpoint = os.getenv("WEB_VALIDATOR_URL")
+    response = requests.get(f"{web_validator_endpoint}/version")
+    validator_version = response.json()["version"]
+    logging.info(f"Validator version: {validator_version}")
+    return validator_version
+
+
+def update_dataset_metadata(
+    dataset_blobs: List[storage.Blob], validator_version: str
+) -> List[str]:
+    """
+    Update the metadata of the dataset blobs - This will trigger the validation report processor
+    :param dataset_blobs: The dataset blobs to update
+    :param validator_version: The version of the validator
+    :return: The list of updated datasets stable ids
+    """
+    max_retry = os.getenv("MAX_RETRY", None)
+    batch_size = int(os.getenv("BATCH_SIZE", 5))
+    sleep_time = int(os.getenv("SLEEP_TIME", 5))
+    logging.info(f"Max retry: {max_retry}, Batch size: {batch_size}")
+
+    batch_count = 0
+    updated_datasets = []
+    for blob in dataset_blobs:
+        # Update metadata to trigger the validation report processor
+        metadata = blob.metadata if blob.metadata is not None else {}
+        retry = 0
+        if "retry" in metadata:
```

Comment:
You should consider giving this a slightly more explicit name, like `validation_retry`.
Somebody looking at the metadata of the file in the bucket might not know what `retry` mean. `validation_retry` gives just a bit more information.

Reply:
Resolving this comment since the trigger is now done directly through the function (not with metadata update anymore), so this feature have been removed. We plan to introduce a way to track executions for all functions after MVP (cc: @davidgamez)